lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1369,6 +1369,37 @@ bool ValueLifetimeAnalysis::isWithinLifetime(SILInstruction *Inst) {
   llvm_unreachable("Expected to find use of value in block!");
 }
 
+static bool blockContainsDeallocRef(SILBasicBlock *BB, SILInstruction *DefValue,
```

Comment:
Can you add a comment here describing what you are doing? I would say something like:

```
Searches \p BB backwards from \p FrontierInst to the beginning of the list and returns true if we find a dealloc_ref /before/ we find the instruction that defines our target value.
```

Also, can you rename DefValue => ValueDefiningInst? Then it is really clear.

Reply:
I'll keep DefValue, because it's consistent with ValueLifetimeAnalsis::DefValue